Tests/Realm.Tests/Sync/DataTypeSynchronizationTests.cs

```diff
@@ -236,7 +246,7 @@ public class DataTypeSynchronizationTests : SyncTestBase
         public void Dict_Binary() => TestDictionaryCore(o => o.ByteArrayDict, TestHelpers.GetBytes(10), TestHelpers.GetBytes(15), (a, b) => a.SequenceEqual(b));
 
         [Test]
-        public void Property_Binary() => TestPropertyCore(o => o.ByteArrayProperty, (o, rv) => o.ByteArrayProperty = rv, TestHelpers.GetBytes(5), TestHelpers.GetBytes(10), (a, b) => a!.SequenceEqual(b!));
+        public void Property_Binary() => TestPropertyCore(o => o.ByteArrayProperty, (o, rv) => o.ByteArrayProperty = rv, TestHelpers.GetBytes(5), TestHelpers.GetBytes(10));
```

Comment:
Why was this changed?

Reply:
I adopted the pattern of using `Assert.That` with `Is.EqualTo().Using()` with the `RealmValueWithCollections.RealmValueComparer`. I guess it wraps non-RealmValues in RealmValues, but considered it easier than abstracting the comparison just for this single case.